src/_tests_/Md5Form.test.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable testing-library/no-debugging-utils */
+import { render, screen } from "@testing-library/react"
+import userEvent from '@testing-library/user-event';
+
+import Md5Form from "../components/Md5Form"
+import { getMd5 } from "../providers/md5Provider";
+
+xdescribe('Md5Form Component', () => {
+    it('renders Md5Form Component', () => {
+        render(<Md5Form />)
+        screen.debug()
+    })
+    it('text from input field appears in element with class "data-text"', async () => {
+        // czy async/ await jest konieczny? poniżej zakomentowałam alternatywny test, który wydaje mi się, też spełnia swoją funkcję
+        render(<Md5Form />)
+
+        const input = screen.getByRole('textbox')
+        userEvent.type(input, 'test')
+
+        const span = await screen.findByText('test')
```

Comment:
Odnosząc się do komentarza niżej.

`findBy...` działa tak, że czeka 1000ms (domyślny czas) na wykonanie operacji.
Natomiast `getBy...` jest praktycznie uruchmianay natychmiast.

Jeśli między "1000ms", a "praktycznie natychmiast" przeładuje się `state` to test zadziała. 
Jeśli nie to będzie błąd. Dlatego lepiej wykorzystać `findBy..` z obietnicą i mieć pewność, że będzie ok niż liczyć na szczęście ;P